video_pose_estimation.py

```diff
@@ -4,212 +4,248 @@
+                rightWristAngle = calculate_angle(right_elbow, right_wrist, right_index)
+                rightElbowAngle = calculate_angle(right_shoulder, right_elbow, right_wrist)
+                rightShoulderAngle = calculate_angle(right_hip, right_shoulder, right_elbow)
+                rightHipAngle = calculate_angle(right_knee, right_hip, right_shoulder)
+                rightKneeAngle = calculate_angle(right_ankle, right_knee, right_hip)
+                rightAnkleAngle = calculate_angle(right_foot_index, right_knee, right_hip)
+
+                angles = [leftWristAngle, leftElbowAngle, leftShoulderAngle, leftHipAngle, leftKneeAngle, leftAnkleAngle,
+                      rightWristAngle, rightElbowAngle, rightShoulderAngle, rightHipAngle, rightKneeAngle,
+                      rightAnkleAngle]
+
+                angleStrings = ["leftWristAngle", "leftElbowAngle", "leftShoulderAngle", "leftHipAngle", "leftKneeAngle",
+                            "leftAnkleAngle", "rightWristAngle", "rightElbowAngle", "rightShoulderAngle",
+                            "rightHipAngle", "rightKneeAngle", "rightAnkleAngle"]
+
+                bodyAngles = dict(zip(angleStrings, angles))
+
+            except:
+                print("Halt!")
+                pass
+
+            # landmarks[mp_pose.PoseLandmark.LEFT_SHOULDER.value] gives coordinate
+            # mp_pose.PoseLandmark.LEFT_SHOULDER.value gives index of landmark in the PoseLandmark array
+            # Draw the pose annotation on the image.
+            mp_drawing.draw_landmarks(
+                image, results.pose_landmarks, mp_pose.POSE_CONNECTIONS)
+
+            image = cv2.cvtColor(image, cv2.COLOR_BGR2RGB)
+
+            img = Image.fromarray(image, 'RGB')
+
+            pg_img = pygame.image.frombuffer(img.tobytes(), img.size, img.mode)
+            screen.blit(pg_img, (0, 0))
+            pygame.display.flip()
+
+            if cv2.waitKey(10) & 0xFF == ord('q'):
+                break
+
+        cap.release()
+        cv2.destroyAllWindows()
+    print("")
+    pprint(bodyAngles, width=1)
+    print("")
+
+    for angle in instructorBodyAngle:
+        for angles in bodyAngles:
+            if (instructorBodyAngle[angle] == bodyAngles[angle]) or (
+                    round(abs(instructorBodyAngle[angle] - bodyAngles[angle]), 2) <= 10):
+                print(angle, "is aligned")
+                good_counter += 1
+                break
+            else:
+                print(angle, "is off by", round(abs(instructorBodyAngle[angle] -
+                                                    bodyAngles[angle]), 2), "degrees")
+                off_counter += 1
+                break
+    print(" ")
+    print(good_counter)
+    print(off_counter)
+
+    if good_counter >= 10:
+        print("Job well done!")
+    elif (good_counter >= 5) and (good_counter <= 9):
+        print("Good job, we go again!")
+    else:
+        print("common! this is not it")
+
+
+
+pygame.init()
```

Comment:
This is what gets run when you run this program, it creates the menu system which will run your program when you click "Play"